Revamp the H2P's explanation of night actions

TheHans255 opened the pull request:
# Description

- Swap the orders of Day Phase and Night Phase to more accurately reflect the order they come in
- Explain the anonymous identity mechanic
- Briefly mention smiting
- Move the Night Action Order to a separate page
- Replace the Night Action Order on the main page with expository explanations of the most common actions, including kills, intel, protection, revives, roleblocks, and identity swaps.
- Correct the order of night actions to reflect recent changes
- Discuss how night action conflicts are handled

# Checklist

I have checked the following _(mark as completed with [x])_:

- [x] The content is accurate (and confirmed with the mods)
- [x] Casing and wording of keywords such as role names, factions, etc, is all correct
- [x] The Markdown is valid
- [x] It renders correctly _(follow README for instructions on previewing)_
- [x] The sidebar links are correct and in the right sections _(follow README for instructions on previewing)_

Commit:
Revamp the H2P's explanation of night actions

- Swap the orders of Day Phase and Night Phase to more accurately
reflect the order they come in
- Explain the anonymous identity mechanic
- Briefly mention smiting
- Move the Night Action Order to a separate page
- Replace the Night Action Order on the main page with expository
explanations of the most common actions, including kills, intel,
protection, revives, roleblocks, and identity swaps.
- Correct the order of night actions to reflect recent changes
- Discuss how night action conflicts are handled

Kirschstein commented on src/guides/aaa_general/night-action-order.mdx:
```diff
@@ -0,0 +1,34 @@
+1. Passing of items
+1. Swap Identities (Djinn, Shapeshifter)
+1. Report kills to village
+1. Report revives to village
```
all marked as 1.

TheHans255 replied:
That's a Markdown trick - you can mark everything in the ordinal list as 1 and it will create the right numbering for you.

ShaneMcC replied:
Technically markdown will do the right thing here, but it would be nicer to be numbered correctly in the source

Kirschstein replied:
TIL...

StormPooper replied:
I actually prefer this approach as it avoids mistakes when people add an entry.

TheHans255 replied:
I prefer it as well, as it keeps the Git history cleaner - if someone were to add an entry, they would not need to change the following lines and influence the `git blame` output.

Kirschstein commented on src/guides/aaa_general/night-action-order.mdx:
```diff
@@ -0,0 +1,34 @@
+- Redirects and identity swaps occur in a determinsitic, server-defined order, which may cause them to be unpredictable if two or more such actions
+  target the same player.
+- If a player is roleblocked multiple times, they will receive one message for each roleblock, but receive no additional effect.
+- Protection visits stack - the player is protected by all sources protecting them, and if a player is attacked, all associated effects trigger.
```
Unsure if "stack" is the best word to use here

StormPooper replied:
Agree, it makes it sound like you're protected multiple times.

TheHans255 replied:
Good point - I will adjust that. I think I'll use the word "overlap", as the player can't get any additional benefit from being protected multiple times.

TheHans255 replied:
Corrected to "overlap"

Kirschstein commented on src/guides/aaa_general/night-action-order.mdx:
```diff
@@ -0,0 +1,34 @@
+- If a player is roleblocked multiple times, they will receive one message for each roleblock, but receive no additional effect.
+- Protection visits stack - the player is protected by all sources protecting them, and if a player is attacked, all associated effects trigger.
+- If multiple players attempt to steal from the same target with the same priority of action, no items are stolen.
+- If multiple players successfully kill the same target, all effects attached to all kills trigger.
```
True but not complete.

The kill would be attributed to a single cause, but their attacks would have success after effects triggered

TheHans255 replied:
Good point, will add that. I assume this also affects item thefts as well - I will correct that too.

TheHans255 replied:
Added and corrected

Kirschstein commented on src/guides/aaa_general/aaa_basics.mdx:
```diff
@@ -6,37 +6,39 @@ route: /how-to-play-online-werewolf
+- Actions that swap identities (such as that of the Shapeshifter). Targets of these actions swap their avatar and username assignments - this swap is not made known to anyone besides the targets, and each will appear as their new identities to the rest of the village. These swaps, however, do not exchange any other characteristics of the players, including their factions, night chats, abilities,  ...
 
-Note that all roles get reports on the situation before identity-swapping abilities, meaning that your information is possibly outdated by the time you receive it.
+Night actions are committed simultaneously at the end of the night and are resolved in a mostly deterministic way by the game engine. This order is detailed on the Night Action Order page.
```
Would either link directly to the page or bring it back to this one

TheHans255 replied:
How do I link pages in this engine? The H2P at present contains no links.

StormPooper replied:
Simple Markdown links _should_ suffice, so `[text](/url)`.

TheHans255 replied:
Yep, that works! Thank you.

Kirschstein commented:
TIL...

On Wed, Aug 26, 2020 at 5:44 PM Shane Mc Cormack <notifications@github.com>
wrote:

> *@ShaneMcC* commented on this pull request.
> ------------------------------
>
> In src/guides/aaa_general/night-action-order.mdx
> <https://github.com/werewolv-es/how-to-play/pull/215#discussion_r477441634>
> :
>
> > +
> +# Order of Night Actions
> +
> +The different abilities roles have at night may influence each other. How they do is determined by the order in which the night actions are defined to take place. The order of night actions is:
> +
> +1. Redirects (Succubus)
> +1. Roleblocks (Direwolf, Courtesan, Manacles of Spite)
> +1. Protection visits (Protector, Huntsman, Shaman, Armour, Heavy Shield)
> +1. Most visits (Too many to list: all visits not belonging in another category, including items)
> +1. Item thefts from abilities and potions
> +1. Kills (Killing werewolf (Werewolf/Alphawolf/Shapeshifter), Vampire, Witch, Militia), including resulting item thefts
> +1. Report visits to Stalkers, Harlots, Familiar
> +1. Passing of items
> +1. Swap Identities (Djinn, Shapeshifter)
> +1. Report kills to village
> +1. Report revives to village
>
> Technically markdown will do the right thing here, but it would be nicer
> to be numbered correctly in the source
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/werewolv-es/how-to-play/pull/215#discussion_r477441634>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACRT4F32FETE2HAF6GOLGDSCU3YLANCNFSM4QLOJYOQ>
> .
>

StormPooper commented on src/guides/aaa_general/night-action-order.mdx:
```diff
@@ -0,0 +1,34 @@
+---
+name: Basics
+menu: Night Action Order
```
You'll want to add this menu item to the array in `doczrc.js` or it'll appear at the bottom of the list: https://github.com/werewolv-es/how-to-play/blob/3b7b12e3d741dff988cb63f45be589c761e27e9d/doczrc.js#L10

TheHans255 replied:
Will do.

TheHans255 replied:
Added

StormPooper commented on src/guides/aaa_general/night-action-order.mdx:
```diff
@@ -0,0 +1,34 @@
+---
+name: Basics
+menu: Night Action Order
```
```suggestion
name: Night Action Order
menu: General
```

Commit:
Merge branch 'site' into thehans255/revamp-night-action-h2p

Commit:
Correct name/menu on night-action-order.mdx

Co-authored-by: Daniel Smith <1657182+StormPooper@users.noreply.github.com>

Commit:
Address PR comments

- Correct name/menu on Night Action Order
- Protection visits "overlap", not "stack"
- Link directly to Night Action Order page on Basics
- Clarify that in multiple kills, a single "cause" is attributed

Also make the formatting on the night action summaries more snappy

Commit:
Merge branch 'thehans255/revamp-night-action-h2p' of github.com:TheHans255/werewolves-how-to-play into thehans255/revamp-night-action-h2p

Kirschstein committed:
Merge branch 'site' into thehans255/revamp-night-action-h2p
